docs/book/src/roadmap.md

```diff
@@ -3,32 +3,37 @@
 This roadmap is a constant work in progress, subject to frequent revision. Dates are approximations.
 
 
-## v0.3.7 (v1alpha3+) ~ June/July 2020
+## v0.4 (v1alpha4) ~ Q1 2021
 
 |Area|Description|Issue/Proposal|
 |--|--|--|
-|Testing|E2E Test plan|[Spreadsheet](https://docs.google.com/spreadsheets/d/1uB3DyacOLctRjbI6ov7mVoRb6PnM4ktTABxBygt5sKI/edit#gid=0)
-|Testing|Enable webhooks in integration tests|[#2788](https://github.com/kubernetes-sigs/cluster-api/issues/2788)|
-|Control Plane|KubeadmControlPlane robustness|[#2753](https://github.com/kubernetes-sigs/cluster-api/issues/2753)|
-|Control Plane|KubeadmControlPlane adoption|[#2214](https://github.com/kubernetes-sigs/cluster-api/issues/2214)|
-|Extensibility|Clusterctl library should support extensible templating|[#2339](https://github.com/kubernetes-sigs/cluster-api/issues/2339)|
-|Cluster Lifecycle|ClusterResourceSet experiment|[#2395](https://github.com/kubernetes-sigs/cluster-api/issues/2395)|
-|Core Improvements|Library to watch remote workload clusters|[#2414](https://github.com/kubernetes-sigs/cluster-api/issues/2414)|
-|API, UX|Support and define conditions on cluster api objects|[#1658](https://github.com/kubernetes-sigs/cluster-api/issues/1658)|
-|Extensibility, Infrastructure|Support spot instances|[#1876](https://github.com/kubernetes-sigs/cluster-api/issues/1876)|
-|Extensibility|Machine pre-deletion hooks|[#1514](https://github.com/kubernetes-sigs/cluster-api/issues/1514)|
-
-## v0.4 (v1alpha4) ~ Q4 2020
-
-|Area|Description|Issue/Proposal|
-|--|--|--|
-|UX, Bootstrap|Machine bootstrap failure detection|[#2554](https://github.com/kubernetes-sigs/cluster-api/issues/2554)|
+|Providers|Move to a single manager watching all namespaces for each provider|[#3042](https://github.com/kubernetes-sigs/cluster-api/issues/3042)
+|Clusterctl|Redefine the scope of clusterctl move|[#3354](https://github.com/kubernetes-sigs/cluster-api/issues/3354)
 |Extensibility|Support pluggable machine load balancers|[#1250](https://github.com/kubernetes-sigs/cluster-api/issues/1250)|
 |Tooling Improvements| Define clusterctl inventory specification & have providers implement it|TBA|
 |Core Improvements|Move away from corev1.ObjectReference|[#2318](https://github.com/kubernetes-sigs/cluster-api/issues/2318)|
 |Dependency|Kubeadm v1beta2 types and support|[#2769](https://github.com/kubernetes-sigs/cluster-api/issues/2769)|
+|UX, Bootstrap|Machine bootstrap failure detection|[#2554](https://github.com/kubernetes-sigs/cluster-api/issues/2554)|
```

Comment:
The outcome of previous discussions was that bootstrap failure detection couldn't happen in CAPI because it would have to use infrastructure specific mechanisms but the issue we should consider for v1alpha4 is https://github.com/kubernetes-sigs/cluster-api/issues/3716 (Add a signal for bootstrap success to the bootstrap provider contract)

Reply:
cc @ncdc @randomvariable, are we ok deferring the work on the node agent entirely here and focusing on Cecile's proposal above?

Reply:
If we want to work on a node agent in parallel that's great, but I think it's slightly different than that original bootstrap failure detection issue (which is closed btw) so it should get its own issue

Reply:
I'm in favor of proceeding with 3716 for the time being, as generic bootstrap failure detection is difficult or impossible to do